.github/workflows/e2e_tests.yml

```
@@ -0,0 +1,183 @@
+      - name: Start client dev server
+        working-directory: ./client
+        run: npx webpack serve --env NODE_ENV=development &
+        env:
+          SERVER_HOST: http://localhost:8080
+          KEYCLOAK_HOST: http://localhost:8081
+          KEYCLOAK_REALM_NAME: thesis-management
+          KEYCLOAK_CLIENT_ID: thesis-management-app
+
+      # Wait for both services to be ready
+      - name: Wait for server to be ready
+        run: |
```

actionlint (1.7.11)

[error] shellcheck reported issue in this script: SC2086:info:7:8: Double quote to prevent globbing and word splitting

(shellcheck)